Revit_Engine/Modify/SortEndpoints.cs

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using System.ComponentModel;
+
+namespace BH.Engine.Adapters.Revit
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Flip the input line if its start point is higher than its endpoint.")]
+        [Input("line", "Line to check.")]
+        [Output("line", "The original line if its start point is already below its endpoint. Otherwise, the line flipped.")]
+        public static Line SortEndpoint(this Line locationLine)
+        {
+            if (locationLine.Start.Z == locationLine.End.Z)
+            {
+                BH.Engine.Base.Compute.RecordError(string.Format("Line's start and end points have the same elevation.}"));
+                return null;
+            }
+            else if (locationLine.Start.Z > locationLine.End.Z)
+            {
+                BH.Engine.Base.Compute.RecordNote(string.Format("The input line's bottom was above its top. This line has been flipped to resolve the issue."));
+                locationLine = locationLine.Flip();
+            }
+            return locationLine;
+        }
```

Comment:
Thanks @pawelbaran, I've made the changes but since we shouldn't modify the original Bhom column object's location line, I needed to add the same geometry check to `SetLocation`. This led to duplicated messages:

![image](https://user-images.githubusercontent.com/102604891/162939395-30503eb0-dc4e-48ac-90ee-ddb58a866803.png)

Can we instead separate the checking of line endpoints to a new `SortEndpoints` method under Revit_Engine\Modify? I can then call it from both `SetLocation` and `FamilyInstance`?

Reply:
I see two options here @vietle-bh:
- Use exactly same note in both create and update (identical warnings are suppressed, only one is shown in the UI)
- Create a new method as you suggest - but then please be mindful that it needs to highlight the fact that it is used to sort column's end points to avoid misuse (and the best way to highlight is to embed the information in the method name). Also I would rather put the method in Revit_Core_Engine instead of Revit_Engine - the latter is reflected in all UIs, resulting in the internally used method being available for example in GH.

Both solutions work for me, please feel free to choose the one you prefer 👍

Reply:
Thanks @pawelbaran! I implemented the 2nd option in an attempt to stay DRY :relieved: